### PR TITLE
Generate OLM bundle for devfile registry operator

### PR DESCRIPTION
**Pull request by johnmcollier**
Fixes https://github.com/devfile/api/issues/395

This PR generates the OLM bundles for the Devfile Registry Operator.
- Creates the generated OLM bundles and corresponding bundle image for the Devfile Registry Operator
   - Required bumping the operator's `PROJECT` version from 3-alpha to 3
- Bundle image is named `quay.io/devfile/registry-operator-bundle`
- Updates the CI workflows to build the operator bundle, as well as to push the bundle image to quay.io
- Some manual tweaks to the ClusterServiceVersion (CSV) file
    - Added a brief name and description for the operator. We can add to it later as needed, so I just put a stub description for now, since we aren't onboarding to OperatorHub just yet
    - Added a sample DevfileRegistry CR example to the CSV
    - Added the devfile logo (in base64 form). 
    - Listed myself as the maintainer and listed Red Hat as the "provider"
    - Added the following keywords: - devfile, devfiles, registry, cloud-native, stack, odo

Once this is merged and we have bundle images push to quay.io/devfile/registry-operator-bundle, I'll update the OpenShift CI to test using the OLM bundle instead.

### CODE REVIEW

File: bundle/manifests/registry-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  - supported: false
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  keywords:
+  - devfile
+  - devfiles
+  - registry
+  - cloud-native
+  - stack
```
Comment (maysunfaisal) [inline bundle/manifests/registry-operator.clusterserviceversion.yaml:221]: other suggestions are openshift, microservice

Comment (johnmcollier) [reply bundle/manifests/registry-operator.clusterserviceversion.yaml:221]: Good suggestions.

Comment (johnmcollier) [reply bundle/manifests/registry-operator.clusterserviceversion.yaml:221]: 👍

**Review by maysunfaisal — APPROVED** [review]

**Review by GeekArthur — APPROVED** [review]
LGTM

**Pull request merged**